tests/flow/test_list.py

```diff
@@ -143,3 +143,633 @@ def test03_null_list_function_inputs(self):
+            redis_graph.query(query)
+            query = """MATCH (a:X), (b:Y) create (a)-[r:R]->(b)"""
+            redis_graph.query(query)
+            query = """MATCH ()-[e]->() RETURN toBooleanList(e)"""
+            actual_result = redis_graph.query(query)
+            self.env.assertTrue(False)
+        except ResponseError as e:
+            self.env.assertContains("Type mismatch: expected List but was Edge", str(e))
+            query = """MATCH (a:X),(b:Y) DELETE a, b"""
+            redis_graph.query(query)
+
+        try:
+            query = """CREATE (a:X), (b:Y)"""
+            redis_graph.query(query)
+            query = """MATCH (a:X), (b:Y) CREATE (a)-[r:R]->(b)"""
+            redis_graph.query(query)
+            query = """MATCH p=()-[]->() RETURN toBooleanList(p)"""
+            actual_result = redis_graph.query(query)
+            self.env.assertTrue(False)
+        except ResponseError as e:
+            self.env.assertContains("Type mismatch: expected List but was Path", str(e))
+            query = """MATCH (a:X),(b:Y) DELETE a, b"""
+            redis_graph.query(query)
+
+        # List of node input should return list of NULL
+        query = """CREATE (a:X)"""
+        redis_graph.query(query)
+        query = """MATCH (n) RETURN toBooleanList([n])"""
+        expected_result = [[None]]
+        actual_result = redis_graph.query(query)
+        self.env.assertEquals(actual_result.result_set[0], expected_result)
+        query = """MATCH (a:X) DELETE a"""
+        redis_graph.query(query)
+
+        # # List of edge input should return list of NULL
+        query = """CREATE (a:X), (b:Y)"""
+        redis_graph.query(query)
+        query = """MATCH (a:X), (b:Y) CREATE (a)-[e:R]->(b)"""
+        redis_graph.query(query)
+        query = """MATCH ()-[e]->() RETURN toBooleanList([e])"""
+        expected_result = [[None]]
+        actual_result = redis_graph.query(query)
+        self.env.assertEquals(actual_result.result_set[0], expected_result)
+        query = """MATCH (a:X),(b:Y) DELETE a, b"""
+        redis_graph.query(query)
+
+        # List of path input should return list of NULL
+        query = """CREATE (a:X), (b:Y)"""
+        redis_graph.query(query)
+        query = """MATCH (a:X), (b:Y) create (a)-[e:R]->(b)"""
+        redis_graph.query(query)
+        query = """MATCH p=()-[]->() RETURN toBooleanList([p])"""
+        expected_result = [[None]]
+        actual_result = redis_graph.query(query)
+        self.env.assertEquals(actual_result.result_set[0], expected_result)
+        query = """MATCH (a:X),(b:Y) DELETE a, b"""
+        redis_graph.query(query)
+
+        # Test without input argument
+        # try:
+        #     query = """RETURN toBooleanList()"""
+        #     redis_graph.query(query)
+        #     self.env.assertTrue(False)
+        # except ResponseError as e:
+        #     self.env.assertContains("Received 0 arguments to function 'toBooleanList', expected at least 1", str(e)) 
```

Comment:
Why is this test commented out ?

Reply:
This test is generating this an unhandled exception, which is reported by the memcheck test.
```
        test_list:testList.test04_toBooleanList:
	[ERROR]
	Unhandled exception: Connection closed by server
```
I have not found the reason, I need to do more tests.
It seems like the argv[0] is being read, even when there is no input arguments, because this is reported by valgrid log.
```
==8855== Conditional jump or move depends on uninitialised value(s)
==8855==    at 0x5E384C9: AR_TOBOOLEANLIST (list_funcs.c:135)
==8855==    by 0x5E31954: _AR_EXP_EvaluateFunctionCall (arithmetic_expression.c:473)
==8855==    by 0x5E323F7: _AR_EXP_Evaluate (arithmetic_expression.c:556)
==8855==    by 0x5E323F7: AR_EXP_Evaluate (arithmetic_expression.c:581)
==8855==    by 0x5E3260B: AR_EXP_ReduceToScalar (arithmetic_expression.c:318)
==8855==    by 0x5E3260B: AR_EXP_ReduceToScalar (arithmetic_expression.c:282)
==8855==    by 0x5E34979: AR_EXP_FromASTNode (arithmetic_expression_construct.c:834)
==8855==    by 0x5E61D07: _BuildProjectionExpressions (build_projection_ops.c:91)
==8855==    by 0x5E61E0E: _buildProjectionOps (build_projection_ops.c:147)
==8855==    by 0x5E61E0E: buildReturnOps (build_projection_ops.c:215)
==8855==    by 0x5E4B556: ExecutionPlan_PopulateExecutionPlan (execution_plan.c:42)
==8855==    by 0x5E4B6C8: _process_segment (execution_plan.c:153)
==8855==    by 0x5E4B6C8: _process_segments (execution_plan.c:198)
==8855==    by 0x5E4B6C8: NewExecutionPlan (execution_plan.c:336)
==8855==    by 0x5E47076: ExecutionCtx_FromQuery (execution_ctx.c:117)
==8855==    by 0x5E46B14: _query (cmd_query.c:311)
==8855==    by 0x5E96B5E: thread_do (thpool.c:382)
==8855==  Uninitialised value was created by a stack allocation
==8855==    at 0x5E31554: _AR_EXP_EvaluateFunctionCall (arithmetic_expression.c:430)
```

Reply:
With the changes done in [4e25c0a](https://github.com/RedisGraph/RedisGraph/pull/2605/commits/4e25c0a142f502ac1430e93a07724a27693a7627), tests ran without problems.
The function _AR_EXP_ValidateArgsCount was modified to return a boolean, and it was called in _AR_EXP_EvaluateFunctionCall, if there is an error, the evaluation returns error to avoid the execution of the expressions.